### PR TITLE
Re-injecting the files in a new branch

### PR DESCRIPTION
**Pull request by vplauzon**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

File: quickstarts/microsoft.kusto/kusto-event-hub/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
```
Comment (bmoore-msft) [inline quickstarts/microsoft.kusto/kusto-event-hub/deploy.sh:1]: remove this file

File: quickstarts/microsoft.kusto/kusto-event-hub/transpile.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
```
Comment (bmoore-msft) [inline quickstarts/microsoft.kusto/kusto-event-hub/transpile.sh:1]: remove this file

File: quickstarts/microsoft.kusto/kusto-event-hub/main.bicep [hunk 3]
```diff
@@ -0,0 +1,111 @@
+@description('Location for all resources')
```
Comment (bmoore-msft) [inline quickstarts/microsoft.kusto/kusto-event-hub/main.bicep:1]: add a line break between each param declaration

File: quickstarts/microsoft.kusto/kusto-event-hub/main.bicep [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+  resource kustoDb 'databases' = {
+    name: databaseName
+    location: location
+    kind: 'ReadWrite'
+
+    resource kustoScript 'scripts' = {
+      name: 'db-script'
+      properties: {
+        scriptContent: loadTextContent('script.kql')
+        forceUpdateTag:  'abc'
```
Comment (bmoore-msft) [inline quickstarts/microsoft.kusto/kusto-event-hub/main.bicep:57]: if this is needed to force script execution, use a valid seed.  If it's not needed, remove it

File: quickstarts/microsoft.kusto/kusto-event-hub/main.bicep [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+      properties: {}
+    }
+  }
+}
+
+resource cluster 'Microsoft.Kusto/clusters@2022-02-01' = {
+  name: clusterName
+  location: location
+  sku: {
+    'name': 'Standard_D12_v2'
```
Comment (bmoore-msft) [inline quickstarts/microsoft.kusto/kusto-event-hub/main.bicep:40]: make this a parameter (and set the defaultValue)

File: quickstarts/microsoft.kusto/kusto-event-hub/main.bicep [hunk 6]
```diff
@@ -0,0 +1,111 @@
+@description('Location for all resources')
+param location string = resourceGroup().location
+@description('Name of the cluster')
+param clusterName string = 'kusto${uniqueString(resourceGroup().id)}'
+@description('Name of the database')
+param databaseName string = 'kustodb'
+@description('Name of Event Hub\'s namespace')
+param eventHubNamespaceName string = 'eventHub${uniqueString(resourceGroup().id)}'
```
Comment (bmoore-msft) [inline quickstarts/microsoft.kusto/kusto-event-hub/main.bicep:8]: are eventhub names globally unique?

File: quickstarts/microsoft.kusto/kusto-event-hub/main.bicep [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+    }
+  }
+}
+
+
+//  We need to authorize the cluster to read the event hub by assigning the role
+//  "Azure Event Hubs Data Receiver"
+//  Role list:  https://docs.microsoft.com/en-us/azure/role-based-access-control/built-in-roles
+var dataReceiverId = 'a638d3c7-ab3a-418d-83e6-5f17a39d4fde'
+var fullDataReceiverId = '/subscriptions/${subscription().subscriptionId}/providers/Microsoft.Authorization/roleDefinitions/${dataReceiverId}'
```
Comment (bmoore-msft) [inline quickstarts/microsoft.kusto/kusto-event-hub/main.bicep:96]: use 
```
subscriptionResourceId('Microsoft.Authorization/roleDefinitions', dataReceiverId)
```
for the resourceId

File: quickstarts/microsoft.kusto/kusto-event-hub/main.bicep [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+var eventHubRoleAssignmentName = '${resourceGroup().id}${cluster.name}${dataReceiverId}${eventHubNamespace::eventHub.name}'
+
+resource clusterEventHubAuthorization 'Microsoft.Authorization/roleAssignments@2021-04-01-preview' = {
+  name: '${guid(eventHubRoleAssignmentName)}'
```
Comment (bmoore-msft) [inline quickstarts/microsoft.kusto/kusto-event-hub/main.bicep:100]: remove the guid function from here and put it on the variable

File: quickstarts/microsoft.kusto/kusto-event-hub/main.bicep [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+  }
+}
+
+
+//  We need to authorize the cluster to read the event hub by assigning the role
+//  "Azure Event Hubs Data Receiver"
+//  Role list:  https://docs.microsoft.com/en-us/azure/role-based-access-control/built-in-roles
+var dataReceiverId = 'a638d3c7-ab3a-418d-83e6-5f17a39d4fde'
+var fullDataReceiverId = '/subscriptions/${subscription().subscriptionId}/providers/Microsoft.Authorization/roleDefinitions/${dataReceiverId}'
+var eventHubRoleAssignmentName = '${resourceGroup().id}${cluster.name}${dataReceiverId}${eventHubNamespace::eventHub.name}'
```
Comment (bmoore-msft) [inline quickstarts/microsoft.kusto/kusto-event-hub/main.bicep:97]: change the guid seeds here to match the principal, role and scope of the roleAssignment

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: quickstarts/microsoft.kusto/kusto-event-hub/main.bicep (hunk 6 above)
Comment (vplauzon) [reply quickstarts/microsoft.kusto/kusto-event-hub/main.bicep:8]: Yes as they map to a domain name (maybe region-global)

File: quickstarts/microsoft.kusto/kusto-event-hub/transpile.sh (hunk 2 above)
Comment (vplauzon) [reply quickstarts/microsoft.kusto/kusto-event-hub/transpile.sh]: Done

File: quickstarts/microsoft.kusto/kusto-event-hub/main.bicep (hunk 3 above)
Comment (vplauzon) [reply quickstarts/microsoft.kusto/kusto-event-hub/main.bicep:1]: Done.

File: quickstarts/microsoft.kusto/kusto-event-hub/main.bicep (hunk 5 above)
Comment (vplauzon) [reply quickstarts/microsoft.kusto/kusto-event-hub/main.bicep]: Done.

File: quickstarts/microsoft.kusto/kusto-event-hub/main.bicep (hunk 7 above)
Comment (vplauzon) [reply quickstarts/microsoft.kusto/kusto-event-hub/main.bicep]: Done.

File: quickstarts/microsoft.kusto/kusto-event-hub/main.bicep (hunk 9 above)
Comment (vplauzon) [reply quickstarts/microsoft.kusto/kusto-event-hub/main.bicep:108]: Done.

File: quickstarts/microsoft.kusto/kusto-event-hub/main.bicep (hunk 8 above)
Comment (vplauzon) [reply quickstarts/microsoft.kusto/kusto-event-hub/main.bicep]: Done.

File: quickstarts/microsoft.kusto/kusto-event-hub/main.bicep (hunk 4 above)
Comment (vplauzon) [reply quickstarts/microsoft.kusto/kusto-event-hub/main.bicep]: Done (removed)

**Comment by vplauzon** [comment]
Thank you Brian!

**Pull request merged**